packages/list-store/src/search-params.ts

```diff
@@ -73,16 +96,11 @@ function fallback<T>(value: T) {
   return z.any().transform(() => value);
 }
 
-const searchOptionsFilterSchema = z
-  .string()
-  .optional()
-  .transform(filterValue => filterValue?.split(',').filter(id => !!id))
-  .pipe(z.array(z.string()).optional().default([]));
-
-interface FromSearchParamsToSearchOptionsProps {
+export interface FromSearchParamsToSearchOptionsProps {
   // `searchParams` is a key value object based on the url search params.
   searchParams: {
-    [filter: string]: string;
+    // Search params are always strings or arrays of strings. Note that numbers are also strings, so we need to transform them to numbers.
```

Comment:
I find the addition `so we need to transform them to numbers.` a little bit confusing - the code doesn't transform strings to numbers? Would it be an option to remove the comment as a whole (because the type definition on line 103 already shows the types allowed)?

Reply:
The transformation is done on line 166. I thought it would be relevant to mention that even if the input is always strings, the output is not. But I will remove this line.


The transformation code. `+value` transforms a string to a number.
```
          } else if (filterKey.type === 'number') {
            acc[filterKey.name] = z
              .string()
              .transform(value => +value)
              .optional()
              .or(fallback(undefined));
          }
```